generic/yajltcl.c

```diff
@@ -851,6 +853,36 @@ yajltcl_yajlObjectObjCmd(ClientData cData, Tcl_Interp *interp, int objc, Tcl_Obj
               gstatus = yajl_gen_string (hand, (unsigned char *)string, len);
               break;
           }
+			
+	  // convert a Tcl lat/lon list to yajl objects
+          case OPT_ADD_LATLON_LIST: {
+            Tcl_Obj **linesPtr;
+            int lines;
+            char *latlon;
+            int len;
+
+            if (arg + 1 >= objc) {
+                Tcl_WrongNumArgs (interp, 1, objv, "list value");
+                return TCL_ERROR;
+            }
+
+            if (Tcl_ListObjGetElements(interp, objv[++arg], &lines, &linesPtr) == TCL_ERROR) {
```

Comment:
Maybe also validate the `lines` is an even number of elements?

Reply:
Added a check for even list.